hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneClientAdapterImpl.java

```diff
@@ -1066,4 +1069,23 @@ public boolean isFSOptimizedBucket() {
     // TODO: Need to refine this part.
     return false;
   }
+
+  @Override
+  public FileChecksum getFileChecksum(Path f, long length) throws IOException {
+    Options.ChecksumCombineMode combineMode =
+        config.getObject(OzoneClientConfig.class).getChecksumCombineMode();
+
+    OFSPath ofsPath = new OFSPath(f);
+    String keyName = ofsPath.getKeyName();
+    if (keyName.length() == 0) {
```

Comment:
Please move the `if (keyName.length() == 0) {` validation inside `OzoneClientUtils.getFileChecksumWithCombineMode()` so that both o3fs and ofs will have the check.

Also, Please make a `Preconditions.checkArgument(length >= 0); `check.